stuf_zds_payments/plugin.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from typing import Any
+
+from django.utils.translation import gettext_lazy as _
+
+from openforms.registrations.contrib.stuf_zds.plugin import (
+    StufZDSRegistration,
+    ZaakOptionsSerializer,
+)
+from openforms.registrations.contrib.stuf_zds.utils import flatten_data
+from openforms.registrations.registry import register
+from openforms.submissions.models import Submission
+from openforms.variables.service import get_static_variables
+from rest_framework import serializers
+from stuf.stuf_zds.client import ZaakOptions
+
+from .client import get_client
+from .registration_variables import register as variables_registry
+
+logger = logging.getLogger(__name__)
+
+PLUGIN_IDENTIFIER = "stuf-zds-create-zaak:ext-utrecht"
+
+
+def default_payment_status_update_mapping() -> list[tuple[str, str]]:
+    return [
+        ("payment_completed", "payment_completed"),
+        (
+            "payment_amount",
+            "payment_amount",
+        ),
+        (
+            "payment_public_order_ids",
+            "payment_public_order_ids",
+        ),
+    ]
+
+
+def prepare_value(value: Any):
+    match value:
+        case bool():
+            return "true" if value else "false"
+        case float():
+            return str(value)
+        case _:
+            return value
+
+
+class ZaakPaymentOptionsSerializer(ZaakOptionsSerializer):
+    payment_status_update_mapping = serializers.ListField(
+        child=serializers.ListField(
+            child=serializers.CharField(), min_length=2, max_length=2
+        ),
+        label=_("payment status update variable mapping"),
+        help_text=_(
+            "This mapping is used to map the variable keys to keys used in the XML "
+            "that is sent to StUF-ZDS. Those keys and the values belonging to them in "
+            "the submission data are included in extraElementen."
+        ),
+        required=False,
+    )
```

Comment:
I wondering (but don't know how well this plays with rjsf) if explicit labels are an option?

```py
class MappingSerializer(serializers.Serializer):
    form_variable = serializers.CharField(...)
    stuf_name = serializers.CharField(...)


class ZaakPaymentOptionsSerializer(...)
    payment_status_update_mapping = MappingSerializer(many=True)
```

I think this would be preferable to reason about named parameters instead of positional arguments.